src/io/unix/iouring.rs

```diff
@@ -0,0 +1,176 @@
+use super::sync::{File as SyncFile, IODriver as SyncDriver};
+use crate::prelude::*;
+use bytes::{Bytes, BytesMut};
+use rio::Rio;
+use std::time::SystemTime;
+
+/// IO driver for file operations with optional async support
+#[derive(Debug, Clone)]
+pub struct IODriver {
+    rio: Option<Arc<Rio>>,
+    sync_driver: SyncDriver,
+}
+
+impl IODriver {
+    /// Create new iodriver with optional async support
+    pub fn new(rio: Option<Rio>) -> Self {
+        Self {
+            rio: rio.map(Arc::new),
+            sync_driver: SyncDriver::default(),
+        }
+    }
+
+    pub(crate) async fn open(&self, path: impl AsRef<Path>) -> IOResult<File> {
+        let sync = self.sync_driver.open(path).await?;
+        File::from_file(sync, self.rio.clone()).await
+    }
+
+    pub(crate) async fn create(&self, path: impl AsRef<Path>) -> IOResult<File> {
+        let sync = self.sync_driver.create(path).await?;
+        File::from_file(sync, self.rio.clone()).await
+    }
+}
+
+impl Default for IODriver {
+    fn default() -> Self {
```

Comment:
I think, it is a bad idea to have `default` for `IoDriver`. It is better to have multiple explicit constructors to make all intentions clear:
1. `new_sync() -> Self`
2. `new_async() -> Result<Self>`
3. `new_async_with_rio(rio: Rio) -> Self`
4. `new_async_fallback_to_sync()`
5. `new()` - calls `new_async_fallback_to_sync()`

Reply:
Why not single new, special for every iodriver, and single default, which should be minimal initialization constructor?